src/main/groovy/com/bmuschko/gradle/docker/utils/DockerThreadContextClassLoader.groovy

```diff
@@ -72,24 +72,30 @@ class DockerThreadContextClassLoader implements ThreadContextClassLoader {
      * @param dockerClientConfiguration Docker client configuration
      * @return DockerClient instance
      */
-    private getDockerClient(DockerClientConfiguration dockerClientConfiguration) {
-        // Create configuration
-        Class dockerClientConfigClass = loadClass('com.github.dockerjava.core.DockerClientConfig')
-        Method dockerClientConfigMethod = dockerClientConfigClass.getMethod('createDefaultConfigBuilder')
-        def dockerClientConfigBuilder = dockerClientConfigMethod.invoke(null)
-        dockerClientConfigBuilder.withUri(dockerClientConfiguration.url)
-
-        if(dockerClientConfiguration.certPath) {
-            dockerClientConfigBuilder.withDockerCertPath(dockerClientConfiguration.certPath.canonicalPath)
+    private def getDockerClient(DockerClientConfiguration clientConfiguration) {
+        Class clientConfigClass = loadClass('com.github.dockerjava.core.DockerClientConfig')
+        Method clientConfigMethod = clientConfigClass.getMethod('createDefaultConfigBuilder')
+
+        def clientConfigBuilder = clientConfigMethod.invoke(null)
+        clientConfigBuilder.withUri(clientConfiguration.url)
+
+        if(clientConfiguration.certPath) {
+            clientConfigBuilder.withDockerCertPath(clientConfiguration.certPath.canonicalPath)
         }
 
-        def dockerClientConfig = dockerClientConfigBuilder.build()
+        def version = client(clientConfigBuilder).versionCmd().exec().apiVersion
+        clientConfigBuilder.withVersion(version)
+
+        return client(clientConfigBuilder)
+    }
+
+    private def client(def configBuilder) {
```

Comment:
Again, no need to return def. We also might want to rename this `createClient`. Looks like there is a bit of code duplication that we may be able to rid ourselves of by passing in parameters to this method.